client/source/js/modules/common/normalise-height-directive.js

```diff
@@ -0,0 +1,122 @@
+define(['angular', 'jquery', 'underscore'], function (angular, $, _) {
+  'use strict';
+
+  return angular.module('app.common.normalise-height', [])
+
+    /**
+     * This directive normalises the height to the tallest element having the
+     * `normalise-height` attribute currently available in the DOM.
+     *
+     * In this example both elements will end up with a height of 200px:
+     *  <div style="height: 120px" normalise-height></div>
+     *  <div style="height: 200px" normalise-height></div>
+     *
+     * Important: Any element that re-draws itself & changes the height
+     * during this process must trigger a jQuery `draw` event for the new
+     * height to be taken into account.
+     */
+    .directive('normaliseHeight', function(normalisingHeightStore) {
+      return {
+        restrict : 'A',
+        terminal : true,
+        link : function(scope, element) {
+          normalisingHeightStore.registerElement(element);
+
+          scope.$on("$destroy", function() {
+            normalisingHeightStore.unregisterElement(element);
+          });
+        }
+      };
+    })
+    .factory('normalisingHeightStore', function() {
+      var maxHeight = 0;
+      var elements = [];
+
+      /**
+       * Returns the max height of all existing elements.
+       *
+       * In case the list is empty 0 is returned.
+       */
+      var _maxHeightOfElement = function () {
```

Comment:
Two questions:

1. Why the `_`prefix?
2. Why you made `var actionBlah = function ()` what will create all those closures in every run instead of `this.actionBlah = function ()` that would do it once during initialization?